### ISSUE TITLE
Unable to reproduce the results in the paper

### ISSUE THREAD

**Issue by ZuhongLIU** [NVlabs/HOVER#21]
Hello, I try to train the student policy using distill_mask_modes = {"omnih2o": DISTILL_MASK_MODES_ALL["omnih2o"]}. The teacher policy is trained for 120000 iterations and student policy is trained for 100000 iterations. Here are my results by following the training and eval scripts. 

`All:  mpjpe_g: 148.483          mpjpe_l: 89.959                 mpjpe_pa: 70.885                accel_dist: 3.113               vel_dist: 8.174                 upper_body_joints_dist: 0.208           lower_body_joints_dist: 0.217           root_height_error: 0.068                root_vel_tracking_error: 0.300          root_r_error: 0.094             root_p_error: 0.100             root_y_error: 0.251
Succ:  mpjpe_g: 139.726                 mpjpe_l: 87.247                 mpjpe_pa: 69.486                accel_dist: 2.685               vel_dist: 7.297                 upper_body_joints_dist: 0.200           lower_body_joints_dist: 0.214           root_height_error: 0.065                root_vel_tracking_error: 0.257          root_r_error: 0.090             root_p_error: 0.096 `

It seems that the values of the metric are quite far away from those in the paper. Could your please give some suggestion to this problem? Thanks in adance.

**Comment by huihuaNvidia2023** [comment]
Can you please clarify the metrics more? Is this the teacher for the student? What's the success rate?

In terms of success rate
There is a slightly difference between this implementation and the paper. In the paper, the success criteria is `{no illegal contact}`. However, the implementation we have here is `{no illegal concat, gravity vector<threshold, reference motion dist < threshold}`, which is more strict. If you remove the criteria `{reference motion dist < threshold}`, you will see better success rate.

In terms of tracking metrics
Our suggestion is to get the best teacher policy. Internally, we did this by running eval with multiple checkpoints, and choose the best one. As you may noticed, the training is quite noisy.

As a reference, the best teacher metrics we got is 
MPJPE-G: 100.3,   MPJPE-L: 68.6,  Success rate: 96.5%

**Comment by jinseokbae** [comment]
@huihuaNvidia2023 
Thanks for releasing great work and clarification above.

I'm also in trouble with reproducing the result.
Currently, I'm using `amass_phc_filtered.pkl` (released by prior work) as training dataset, and trained teacher policy (46k, num_envs: 16384) which has following results from the evaluation (criteria for success rate is following the implementation, not paper):

```
Success Rate: [0.9422496073](tel:09422496073)
All:  mpjpe_g: 109.034      mpjpe_l: 72.986         mpjpe_pa: 61.833        accel_dist: 2.572       vel_dist: 6.161         upper_body_joints_dist: 0.172       lower_body_joints_dist: 0.2
09      root_height_error: 0.061        root_vel_tracking_error: 0.242      root_r_error: 0.089         root_p_error: 0.097         root_y_error: 0.260
Succ:  mpjpe_g: 107.907         mpjpe_l: 72.601         mpjpe_pa: 61.622        accel_dist: 2.464       vel_dist: 5.934         upper_body_joints_dist: 0.169       lower_body_joints_dist:
 0.208      root_height_error: 0.060        root_vel_tracking_error: 0.222      root_r_error: 0.088         root_p_error: 0.097         root_y_error: 0.256
All Masked:  mpjpe_g: 109.034   mpjpe_l: 72.986     mpjpe_pa: 61.833    accel_dist: 2.572   vel_dist: 6.161
Succ Masked:  mpjpe_g: 107.907  mpjpe_l: 72.601     mpjpe_pa: 61.622    accel_dist: 2.464   vel_dist: 5.934
```
This is slightly worse than the one you have, but I believe this teacher model is also performing fairly well.

Then, I trained a student model without any modification on the released code, and got following results (200k, num_envs: 8192, omnih2o mode):

```
Success Rate: 0.7320285127
All:  mpjpe_g: 154.118      mpjpe_l: 88.162         mpjpe_pa: 70.237        accel_dist: 2.952       vel_dist: 7.681         upper_body_joints_dist: 0.218         lower_body_joints_dist: 0.205       root_height_error: 0.072        root_vel_tracking_error: 0.318        root_r_error: 0.096         root_p_error: 0.100         root_y_error: 0.299
Succ:  mpjpe_g: 136.876         mpjpe_l: 83.454         mpjpe_pa: 68.014        accel_dist: 2.105       vel_dist: 5.993      upper_body_joints_dist: 0.208        lower_body_joints_dist: 0.201       root_height_error: 0.069        root_vel_tracking_error: 0.232        root_r_error: 0.091         root_p_error: 0.095         root_y_error: 0.268
All Masked:  mpjpe_g: 113.496   mpjpe_l: 87.438     mpjpe_pa: 29.337    accel_dist: 4.232   vel_dist: 10.577
Succ Masked:  mpjpe_g: 99.575   mpjpe_l: 79.991     mpjpe_pa: 27.830    accel_dist: 3.205   vel_dist: 8.645
```
Even the statistics calculated only from the successful trajectories are worse (but quite close) than the one in the paper.

Could you give me the insight of the current result?
Also, it would be great if you can share the exact command you've used for the training.

Thanks for your help in advance.

Best,

**Comment by jinseokbae** [comment]
Thanks for the reply!

This is the command I've used for training:

```
${ISAACLAB_PATH}/isaaclab.sh -p scripts/rsl_rl/eval.py \
    --num_envs 1024 \
    --reference_motion_path neural_wbc/data/data/motions/amass_phc_filtered.pkl \
    --teacher_policy.resume_path neural_wbc/data/data/policy/h1-filtered:teacher/amass_filtered_env16384 \
    --teacher_policy.checkpoint model_46000.pt \
    --student_path neural_wbc/data/data/policy/h1-filtered:student/original_student/amass_filtered_env16384_25_05_22_02-03-07 \
    --student_checkpoint final_model.pt \
    --headless --student_player
```

Additionally, the evaluation is made up with omnih2o mask.
